src/site/topic24.rst

```diff
@@ -174,6 +174,96 @@ Minimum & Maximum
 Remove Minimum & Maximum
 ========================
 
+* Finding the minimum and maximum values in the linked binary search tree is relatively simple
+* However, *removing* from the tree adds extra complexity as we need to preserve the binary search tree ordering of our tree
+
+**Remove Minimum**
+
+.. code-block:: java
+    :linenos:
+
+    public T removeMin() {
+        T returnElement = null;
+        if (isEmpty()) {
+            throw new NoSuchElementException();
+        } else if (root.getLeft() == null) {
+            returnElement = root.getData();
+            root = root.getRight();
```

Comment:
I feel like this can be the base case in the recursive method, instead of the left child having a left child of `null`. I may be worng, I'm only half paying attention right now.

Reply:
you're now requiring me to think. this is a problem.

Reply:
Maybe you need parent references in the node for that to work? A `Node<T> min()` would work if nodes had a parent reference. You could also return two nodes, I guess.

Reply:
That *feels* right.

Reply:
Brain is melt.

Reply:
OK, I am convinced that you're right --- we'd need a parent reference. I spend 3mins trying to articulate why, but can't, but I know you're right RE parent ref. 

If the core of the recursive one is "replace current's left child with the left child's right child", the root is an edge case. That's the best words coming out of my fingers for today.

Reply:
All right. If we're concinved it'd require a parent reference, we can leave it as is for now. Managing parent references can be an aside, a revisit or left for the second year course. Maybe open an issue so it's not forgotten.